spinalcordtoolbox/deepseg/inference.py

```diff
@@ -86,11 +86,25 @@ def segment_and_average_volumes(model_paths, input_filenames, options):
     return nii_lst, target_lst
 
 
-def segment_monai(path_model, input_filenames, threshold):
+def segment_monai(path_model, input_filenames, threshold, device="cpu"):
+    # equivalent to `with torch.no_grad()`
+    torch.set_grad_enabled(False)
+
+    # define device
+    if device == "gpu" and not torch.cuda.is_available():
+        logger.warning("GPU not available, using CPU instead")
+        DEVICE = torch.device("cpu")
+    else:
+        DEVICE = torch.device("cuda" if torch.cuda.is_available() and device == "gpu" else "cpu")
+
+    # load model from checkpoint
+    chkp_path = os.path.join(path_model, "best_model_loss.ckpt")
+    net = create_nnunet_from_plans(chkp_path, DEVICE)
```

Comment:
This is a very good question. I'm not sure, to be honest? I assumed it wouldn't, given that the predictor is based on (fixed) trained model weights, but that's definitely an assumption worth prodding at.

Reply:
So, after `net` is initialized using `create_nnunet_from_plans`, it is used in the following calls:

- MONAI: `sliding_window_inference` function call
- nnUNet: `predictor.predict_single_npy_array` method call (plus reading properties `predictor.plans_manager.dataset_name` and `predictor.dataset_json`)

In other words, there are no calls to e.g. data-specific configuration functions per image -- just inference functions. And, I'm fairly certain that for both the MONAI/nnUNet inference functions, the network is treated as a static function: image comes in, image goes out.

Perhaps this would be different if, say, we specified the `segmentation_previous_stage` argument for nnUNet? But, we don't! :)